### DISCUSSION TITLE
MQTTx Client App, Certificate check and root-Certificate from own authority

### DISCUSSION THREAD

**Discussion by Stefan-Pichler** [emqx/MQTTX#912] [Q&A]
Hello Community, Team,

we have a bunch of Servers which all have SSL certificates which are signed by a own root certificate from a own authority.
The root-Certificate is registered as trusted on the local machine operating System.
I try to give the MQTTx Client App a chance, but i am not able to use the strong certificat check. It seems the MQTTx Client App is not able to check the validity of the SSL-Chain because it does not trust the root certificate. Am i right?
I wonder how to register the root-Certificate into MQTTx to be able to check the validity of all of our certificates.

Please, are you able to give me a hint how to trust the own root Certificate?

Thanks a lot,
Stefan

**Comment by ysfscream** [comment]
Hi, sorry for my late reply.

You can select your root certificate when establishing a TLS/SSL connection.

<img width="666" alt="image" src="https://user-images.githubusercontent.com/21299158/158090579-e1de2d6d-63dc-4e66-ae16-45f8fabd8870.png">

ref: https://www.emqx.com/en/blog/emqx-server-ssl-tls-secure-connection-configuration-guide

You can read this blog, I hope it can help you.
